CMakeLists.txt

```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.8.0)
+
+project(pcg-super)
+
+add_subdirectory(include)
+add_subdirectory(sample)
+
+enable_testing()
+add_subdirectory(test-high)
```

Comment:
These should be guarded like so
https://cliutils.gitlab.io/modern-cmake/chapters/testing.html

And building the samples should be an option, perhaps an option for installing them too.

Reply:
I am not sure if you had time to read through the comments in the pull request thread. It's been almost a year since I made this pull request, and hence, I have forgot most of the code base. Yet, these concerns were discussed back then, and the summary could be found at least under [this](https://github.com/imneme/pcg-cpp/pull/44#issuecomment-496876339) comment.

Have you tried, for instance
```bash
~> git clone https://github.com/aytekinar/pcg-cpp /tmp/pcg-cpp
~> cd /tmp/pcg-cpp
/tmp/pcg-cpp> git checkout 43-cmake
/tmp/pcg-cpp> mkdir build install
/tmp/pcg-cpp> cd build
# only install the header and cmake files
/tmp/pcg-cpp/build> cmake -D CMAKE_INSTALL_PREFIX=../install ../include
/tmp/pcg-cpp/build> cmake --build . && cmake --build . --target install
# super-build, i.e., build sample files as well as tests, and install both header, cmake and sample files
/tmp/pcg-cpp/build> cmake --build .
/tmp/pcg-cpp/build> cmake --build . --target install
# if you want to test
/tmp/pcg-cpp/build> ctest
```

Basically, I had followed Daniel Pfeifer's [talk](https://youtu.be/bsXLMQ6WgIk), which is also listed in "Effective Modern CMake" link under https://cliutils.gitlab.io/modern-cmake/.

IMHO, the simplest solution with CMake is _not_ to use options at all. This is also proposed/suggested by many different people. If you do _not_ want to install/test anything, simply make an out-of-source build of `include`. No need to remember any options, etc. If you want to include this project as a submodule, do so and point in your CMake to this include folder (e.g., `add_subdirectory(external/pcg-cpp/include)`).